Contrast-Security-OSS/go-test-bench
# GO-1570 Update docker setup

With the latest changes the Docker setup is updated with:

- `contrast-service` dependency is removed as it's no longer used by the `go-agent` communication
- added `GO_VERSION` env variable so you can configure specific go version (default is `latest`)
- added `CONTRAST_GO_VERSION` env variable so you can configure specific `contrast-go` version (default is `latest`)
- `contrast-go-installer` is used to download and install `contrast-go`
- platform is set to `linux/amd64` to prevent errors with unsupported `arm64` dependencies


#### Testing

- Run `./demo.sh` and observe successful docker build and container running the test app with no errors
- Set a specific go version or contrast-go version and observe success during docker build.
_Note:_ `contrast-go` must use a version that doesn't use contrast-service
